sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+        this.proxyOptions = proxyOptions;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
+    }
+
+    /**
+     * Asynchronously send the HttpRequest
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param context The context of the request, for any additional changes
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        ProgressReporter progressReporter = Contexts.with(context).getHttpRequestProgressReporter();
+        HttpMethod httpMethod = httpRequest.getHttpMethod();
+        if (httpMethod == HttpMethod.PATCH) {
+            return sendPatchViaSocket(httpRequest);
+        }
+        return handleConnection(httpRequest, progressReporter);
+    }
+
+    /**
+     * Send a PATCH request via a SocketClient
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> SocketClient.sendPatchRequest(httpRequest));
+    }
+
+    /**
+     * Open a connection based on the HttpRequest URL
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param progressReporter (Optional) for reporting progress while writing the request body
+     * @return A Mono containing a HttpUrlConnectionResponse object
+     */
+    private Mono<HttpResponse> handleConnection(HttpRequest httpRequest, ProgressReporter progressReporter) {
+        return Mono.defer(() -> {
+            try {
+                // Make connection
+                URL url = httpRequest.getUrl();
+                HttpURLConnection connection = null;
+
+                if (proxyOptions != null) {
+                    InetSocketAddress address = proxyOptions.getAddress();
+                    if (address != null) {
+                        Proxy proxy = new Proxy(Proxy.Type.HTTP, address);
+                        connection = (HttpURLConnection) url.openConnection(proxy);
+
+                        if (proxyOptions.getUsername() != null && proxyOptions.getPassword() != null) {
+                            String authString = proxyOptions.getUsername() + ":" + proxyOptions.getPassword();
+                            String authStringEnc = Base64.getEncoder().encodeToString(authString.getBytes());
+                            connection.setRequestProperty("Proxy-Authorization", "Basic " + authStringEnc);
+                        }
```

Comment:
`Digest` authentication needs support as well.

Reply:
Digest authentication is nearly there, but we're still working on how to format 'authStringEnc'. Is there any chance we could have some advice on this?